docs/download-upload-clone/legacy-marxan-projects/high-level-design.md

```diff
@@ -0,0 +1,304 @@
+# Uploading legacy Marxan projects - High-level design
+
+Importing legacy projects requires several steps, outlined below.
+
+## Workflow overview
+
+From an API consumer point of view, importing a legacy project is performed
+through three main steps and related endpoints.
+
+### Preparation of project import process
+
+`POST /api/v1/projects/import/legacy`
+
+This step will prepare an empty project shell, as well as an empty scenario
+shell, both of which will be populated later in the import process, unless
+errors are raised while validating user-uploaded data.
+
+The `projectId` of the project being created will be returned, so that the next
+steps can reference the project.
+
+### Data upload
+
+`POST /api/v1/projects/import/legacy/:projectId/data-file`
+
+API clients will need to use this endpoint to upload each of the files that
+users provide.
+
+Each upload will include the actual file alongside a small JSON metadata snippet
+through which the API is informed about which kind of file this is, alongside
+any other metadata needed to import it.
+
+A unique `dataFileId` is returned for each file uploaded successfully.
+
+As API clients send requests to this endpoint, the API keeps the uploaded files
+in a temporary storage area, from which they will be later read and processed.
+
+If a specific kind of file (for example, a planning grid shapefile) is uploaded
+more than once, any previously uploaded copy is discarded and replaced with the
+latest file uploaded. This would be sufficient if a user accidentally uploads
+the wrong file.
+
+In case an uploaded file needs to be fully removed from the set of uploaded
+files (for example, if a user accidentally uploads a `puvspr.dat` file _as if it
+was_ an `output.zip` file), a `DELETE` endpoint is provided, which can be used
+with the `dataFileId` returned on upload:
+
+`DELETE /api/v1/projects/import/legacy/:projectId/data-file/:dataFileId`
```

Comment:
@mbarrenechea does this workflow sound ok/complete from the point of view of the app?